### ISSUE TITLE
After new version is updated it doesn't launch app automatically

### ISSUE THREAD

**Issue by jiakuan** [vslavik/winsparkle#132]
This is more like a question than an issue. I asked this in [StackOverflow][4] but it seems no one knows the answer. Apologies if I should not create an issue for this one. BTW, is there a maillist to discuss this kind of things?

I'm using [WinSparkle][1] (latest version 0.5.3) framework in my Qt C++ app for auto-update. It works perfectly with detecting new version, downloading and installing the new version automatically.

But I found after the new version is installed, it doesn't launch the app automatically, and users have to launch it manually.

I checked the [WinSparkle API reference][2], and didn't find any settings related to this. I think WinSparkle should have built-in support for auto-launching the app after update?

Is there a way in [WinSparkle][1] to handle it in the same way as [Sparkle][3] does?

  [1]: https://github.com/vslavik/winsparkle
  [2]: https://github.com/vslavik/winsparkle/blob/master/include/winsparkle.h
  [3]: https://sparkle-project.org/
  [4]: http://stackoverflow.com/questions/42547680/after-new-version-is-updated-using-winsparkle-it-doesnt-launch-app-automaticall

**Comment by jiakuan** [comment]
I realised the situation might be different between Windows and Mac OS X.

On Windows, perhaps the solution could be to implement this auto-launching inside the installer, e.g. MSI or EXE. Therefore, no need to have this in WinSparkle.

If anyone can confirm that I didn't miss anything available in WinSparkle, I will then close this ticket. Thanks a lot!

**Comment by vslavik** [comment]
> I realised the situation might be different between Windows and Mac OS X.

Indeed. Sparkle has it easy: just replace the files in a bundle; it's also obvious what you launch (the bundle, there's exactly one launchable thing). Neither is true on Windows. At the moment, it's your responsibility to fix your installer to relaunch (or have an option to launch, checked by default) after it's done installing.

That said, *ideally* there would be a mechanism to detect if the installer finished and didn't re-launch, and/or a silent mode where this can be under WinSparkle's control, as well as a Sparkle-like update mechanism where files are updated from a ZIP and re-launching is managed by WinSparkle. Any PRs to that effect, even partial, would always be most welcome.

**Comment by jiakuan** [comment]
I would be nice to be able to update from a ZIP file and re-launch automatically. For now, it's enough to implement that in the Windows installer for me.

I would love to dive in and have a look at the possibility of contributing a PR for this. But unfortunately I'm time poor at the moment.

If you have any guidelines about the best way to implement this new feature (e.g. which class can be created or modified), that would be very helpful.

**Comment by patrickkidd** [comment]
For me, WinSparkle detects and downloads the update properly but then just shows the contents of the downloaded zip file in a temp dir in an explorer window. Is that this same issue or something else? My view is that replacing and auto-launching the new version of an exe seems like a core feature to Sparkle/WinSparkle. The WinSparkle docs do not cover this.
